### PR TITLE
fix(gatsby-source-wordpress): Prevent "EADDRINUSE: address already in use 127.0.0.1" error

### PR DESCRIPTION
**Pull request by TylerBarnes**
It looks like the WP data update watcher code was in some cases running multiple times. It was also running before the dev server was ready. Since data updates emit a WEBHOOK_RECEIVED event to Gatsby, this was causing problems when the dev server wasn't yet ready. We were also potentially emitting this while a WEBHOOK_RECEIVED event was already being processed. This RP fixes all of these issues.

### CODE REVIEW

**Comment by TylerBarnes** [comment]
Closes #30308

File: packages/gatsby-source-wordpress/src/steps/source-nodes/update-nodes/content-update-interval.js [hunk 1]
```diff
@@ -108,28 +111,51 @@ const refetcher = async (
...
+     * onCreateDevServer (the node API we're hooking into) is called before the dev server is ready.
+     * Running our logic at that point is problematic because we could end up triggering the WEBHOOK_RECEIVED event before the dev server is ready and this can cause Gatsby to throw errors. So we're hooking into COMPILATION_DONE to avoid that problem.
+     */
+    if (!firstCompilationDone) {
+      firstCompilationDone = true
+
+      // wait a second so that terminal output is more smooth
+      setTimeout(() => {
+        if (verbose) {
+          helpers.reporter.log(``)
```
Comment (veryspry) [inline packages/gatsby-source-wordpress/src/steps/source-nodes/update-nodes/content-update-interval.js:149]: Do we need this?
```suggestion
```

File: packages/gatsby-source-wordpress/src/steps/source-nodes/update-nodes/content-update-interval.js [hunk 2]
```diff
@@ -108,28 +111,51 @@ const refetcher = async (
...
 
-  if (verbose) {
-    helpers.reporter.log(``)
-    helpers.reporter.info(formatLogMessage`Watching for WordPress changes`)
-  }
-
-  refetcher(msRefetchInterval, helpers)
+  helpers.emitter.on(`COMPILATION_DONE`, () => {
+    /**
+     * we only want to start our refetcher helper 1 time after the first COMPILATION_DONE event.
```
Comment (veryspry) [inline packages/gatsby-source-wordpress/src/steps/source-nodes/update-nodes/content-update-interval.js:138]: Thanks for the explanation!

**Review by veryspry — APPROVED** [review]

File: packages/gatsby-source-wordpress/src/steps/source-nodes/update-nodes/content-update-interval.js (hunk 1 above)
Comment (TylerBarnes) [reply packages/gatsby-source-wordpress/src/steps/source-nodes/update-nodes/content-update-interval.js:149]: It's not necessary but improves the terminal output readability a bit. When this is logged out the rest of the logs (the part telling you to open localhost:8000) is more spaced out, so this keeps the formatting even.

**Pull request merged**